rust/feature-flags/src/request_handler.rs

```diff
@@ -148,70 +184,57 @@ fn decode_request(headers: &HeaderMap, body: Bytes) -> Result<FlagRequest, FlagE
 }
 
 /// Evaluate feature flags for a given distinct_id
-/// Returns a map of feature flag keys to their values
-/// If an error occurs while evaluating a flag, it will be logged and the flag will be omitted from the result
+/// - Returns a map of feature flag keys to their values
+/// - If an error occurs while evaluating a flag, it will be logged and the flag will be omitted from the result
 pub async fn evaluate_feature_flags(
+    team_id: i32,
     distinct_id: String,
     feature_flags_from_cache_or_pg: FeatureFlagList,
     database_client: Option<Arc<dyn Client + Send + Sync>>,
     person_property_overrides: Option<HashMap<String, Value>>,
-    // group_property_overrides: Option<HashMap<String, HashMap<String, Value>>>,
+    group_property_overrides: Option<HashMap<String, HashMap<String, Value>>>,
+    groups: Option<HashMap<String, Value>>,
 ) -> FlagsResponse {
-    let mut matcher = FeatureFlagMatcher::new(
+    let group_type_mapping_cache = GroupTypeMappingCache::new(team_id, database_client.clone());
+    let mut feature_flag_matcher = FeatureFlagMatcher::new(
         distinct_id.clone(),
+        team_id,
         database_client,
-        person_property_overrides,
-        // group_property_overrides,
+        Some(group_type_mapping_cache),
+        None,
+        groups,
     );
-    let mut feature_flags = HashMap::new();
-    let mut error_while_computing_flags = false;
-    let feature_flag_list = feature_flags_from_cache_or_pg.flags;
-
-    for flag in feature_flag_list {
-        if !flag.active || flag.deleted {
-            continue;
-        }
-
-        match matcher.get_match(&flag).await {
-            Ok(flag_match) => {
-                let flag_value = if flag_match.matches {
-                    match flag_match.variant {
-                        Some(variant) => FlagValue::String(variant),
-                        None => FlagValue::Boolean(true),
-                    }
-                } else {
-                    FlagValue::Boolean(false)
-                };
-                feature_flags.insert(flag.key.clone(), flag_value);
-            }
-            Err(e) => {
-                error_while_computing_flags = true;
-                error!(
-                    "Error evaluating feature flag '{}' for distinct_id '{}': {:?}",
-                    flag.key, distinct_id, e
-                );
-            }
-        }
-    }
+    feature_flag_matcher
+        .evaluate_feature_flags(
+            feature_flags_from_cache_or_pg,
+            person_property_overrides,
+            group_property_overrides,
+        )
+        .await
+}
 
-    FlagsResponse {
-        error_while_computing_flags,
-        feature_flags,
-    }
+fn decompress_gzip(compressed: Bytes) -> Result<Bytes, FlagError> {
```

Comment:
You're gonna want to be a little careful to protect against zip-bombs here - I wouldn't bother fixing it now, but I'd add a todo before you go to prod (you can see how this is done in `capture`, and maybe even break that out into something in `common`, but yeah, don't both for now, but add a todo or something)

Reply:
TODO added, good idea on extracting shared request handler methods out to `/common`.